docs/usage/custom-dns.md

```diff
@@ -0,0 +1,51 @@
+# Custom DNS Configuration
+
+Gardener provides Kubernetes-Clusters-As-A-Service where all the system components (e.g., kube-proxy, networking, dns, ...) are managed. 
+As a result, Gardener needs to ensure and auto-correct additional configuration to those system components to avoid unnecaessary down-time. 
+
+In some cases, auto-correcting system components can prevent users from deploying applications on top of the cluster that requires bits of customization, DNS configuration can be a good example. 
+
+To allow for customizations for DNS configuration (that could potentially lead to downtime) while having the option to "undo", we utilize the `import` plugin from CoreDNS [1].
+which enables in-line configuration changes. 
+
+## How to use
+To customize your CoreDNS cluster config, you can simply edit a `configmap` named `custom-dns` in the `kube-system`
+namespace. By editing, this `configmap`, you are modifying CoreDNS configuration, therefore care is advised. 
+
+For example, to apply new config to CoreDNS that would point all `.global` DNS requests to another DNS pod, simply edit the configuration as follows: 
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: coredns-custom
+  namespace: kube-system
+data:
+  istio.server: |
+    global:8053 {
+            errors
+            cache 30
+            forward . 1.2.3.4
+        }
+```
+
+It is important to have the `configmap` key ending with `*.server`. Once this `configmap` is applied, you can 
+roll-out your CoreDNS deployment using 
+> kubectl -n kube-system rollout restart deploy coredns
```

Comment:
```suggestion
\```bash
$ kubectl -n kube-system rollout restart deploy coredns
\```
```

This is the format that is usually used in this repo. If you apply the suggestion, please remove the escape character `\`

Reply:
ok